lib/processor/index.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+const UploadedFile = require('../uploaded-file');
+const logger = require('../logger');
+
+// aliases
+exports.download = require('./download');
+exports.validate = require('./validate');
+exports.uploadAndCopy = require('./upload');
+
+// do all the things
+exports.work = (audioEvent) => {
+  let file = new UploadedFile(audioEvent);
+
+  // process file - catch most errors
+  return exports.download(audioEvent.audioUploadPath).then(data => {
+    file.setDownloaded(data);
+    return Q.all([
+      exports.validate(file.localPath),
```

Comment:
is it validating? more like analyzing now, or probing (that sounds wrong)

Reply:
Yeah, you're right.  It doesn't ever "fail" validation, so it's more like "detection" now.  I'd rather not rename it here, since there's already so much going on.  But I could rename callback `valid` key, if you want?  (Or add a new one, to maintain backwards compatibility?)